src/main/kotlin/no/nav/system/rule/dsl/inspections/XmlDebug.kt

```diff
@@ -0,0 +1,87 @@
+package no.nav.system.rule.dsl.inspections
+
+import no.nav.system.rule.dsl.AbstractRuleComponent
+import no.nav.system.rule.dsl.AbstractRuleflow
+import no.nav.system.rule.dsl.Predicate
+import no.nav.system.rule.dsl.Rule
+import no.nav.system.rule.dsl.rettsregel.AbstractSubsumtion
+import no.nav.system.rule.dsl.rettsregel.Faktum
+
+/**
+ * Lists the complete tree of [AbstractRuleComponent] in XML format.
+ */
+
+fun AbstractRuleComponent.xmlDebug(): String {
+    val debugString = StringBuilder()
+    inspect(this, debugString, 0)
+    return debugString.toString().trim()
+}
+
+private fun inspect(arc: AbstractRuleComponent, debugString: StringBuilder, level: Int) {
+    debugString.append(" ".repeat(level * 2))
+    var tagName = arc.name()
+    val relevantChildren = arc.children.filterNot { it is Faktum<*> }
```

Comment:
filteret her burde være `it.isLeafPairSubsumtion()`. Håper testen fortsatt er fornøyd?